ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java

```diff
@@ -379,6 +387,11 @@ public boolean doNext(WritableComparable key, Writable value) throws IOException
       Class<? extends InputFormat> formatter = currDesc.getInputFileFormatClass();
       Utilities.copyTableJobPropertiesToConf(currDesc.getTableDesc(), job);
       InputFormat inputFormat = getInputFormatFromCache(formatter, job);
+      if(inputFormat instanceof HiveSequenceFileInputFormat) {
+        // input format could be cached, in which case we need to reset the list of files to fetch
```

Comment:
lets also add in comment that format could be cached because of resultset cache

Reply:
@ashutoshc I don't think result cache caches input format. This caching has been there way before result cache so I don't think this comment will be correct.

Reply:
I see. What is this caching is then for?

Reply:
I am not sure, i noticed ptests failing because of this so probably input format is cached by HS for previous queries.